main.go

```diff
@@ -70,7 +71,10 @@ func initArgparser() {
 			os.Exit(1)
 		}
 	}
+}
 
+// parses and validates the arguments
+func parseArguments() {
```

Comment:
SonarCloud complains about the complexity of this method but I would prefer to not refactor this now